mipengine/node/monetdb_interface/monet_db_connection.py

```diff
@@ -73,22 +73,26 @@ def execute_with_result(self, query: str) -> List:
         self._connection.commit()
 
         with self.cursor() as cur:
-            cur.execute(query)
+            cur.executemany(query, parameters) if many else cur.execute(
+                query, parameters
+            )
             result = cur.fetchall()
             return result
 
-    def execute(self, query: str):
+    def execute(self, query: str, parameters=None, many=False):
         """
         Executes statements that don't have a result. For example "CREATE,DROP,UPDATE".
         And handles the *Optimistic Concurrency Control by giving each call X attempts
-        if they fail with pymonetdb.exceptions.IntegrityError .
+        if they fail with pymonetdb.exceptions.IntegrityError.
         *https://www.monetdb.org/blog/optimistic-concurrency-control
         """
 
         for _ in range(OCC_MAX_ATTEMPTS):
             with self.cursor() as cur:
                 try:
-                    cur.execute(query)
+                    cur.executemany(query, parameters) if many else cur.execute(
```

Comment:
@apmariglis @jassak @KFilippopolitis 
I just wanted to note here that the whole `executemany` logic will only be used by `debug` code. 

Code that will never be part of actual production running code. Also in the future there is no scenario that we want to insert many records in the database.

Maybe we shouldn't pollute already complicated production code for a development/debug edge case.

It's not important though, just my 2 cents.